### ISSUE TITLE
Add a private flag so generated structs aren't visible outside the package

### ISSUE THREAD

**Issue by rikonor** [alanctgardner/gogen-avro#16]
Hi,

I noticed that the generated code will contain many exported things that don't necessarily need to be exported.

A few examples are:
```
DeserializeAvroContainerBlock
DeserializeAvroContainerHeader
AvroContainerBlock
AvroContainerHeader
Magic
Sync
```

Should those be left as is or possibly made private?

**Comment by alanctgardner** [comment]
I agree that these should be private, unfortunately because of how they're generated it would be difficult. I'm wondering if adding some godoc comments to these (basically saying "don't use this") would be a good solution short-term.

**Comment by alanctgardner** [comment]
With the most recent refactor to how container files are generated, I think this issue is resolved - these types are all isolated in the container package.

**Comment by rikonor** [comment]
Great, thanks a ton.

That said, though, even though they are in `container`, aren't they still exported?

**Comment by alanctgardner** [comment]
That's correct, they're still exported from `container` right now, which is less than ideal. I'd be open to adding a flag like `--private` that doesn't export the generated structs for this kind of case.

**Comment by rikonor** [comment]
Ah, that's interesting. I did not realize that `container` has generated files. Is that necessary, though? I guess it is a pretty elegant solution. That said, this is a very minor issue overall.

**Comment by alanctgardner** [comment]
Doing a bit of cleanup on this repo - I think the right solution is to add a private flag so structs can be generated as package-private. Then we could use that flag to clean up the container API. I've renamed the ticket and added some tags to clarify.

**Comment by JensRantil** [comment]
Why was reopened?

**Comment by jeroenrinzema** [comment]
@actgardner maybe a property like `golang.private` could work? As we did in #77?
